src/navigations/root-stack.js

```diff
@@ -152,182 +153,318 @@ const AuthenticatedNavigator = () => {
+          )}
           options={{headerShown: false}}
         />
         <AuthenticatedStack.Screen
           name="OtherProfileReplyComment"
-          component={OtherProfileReplyComment}
+          component={(props) => (
+            <KeyboardWrapper>
+              <OtherProfileReplyComment {...props} />
+            </KeyboardWrapper>
+          )}
           options={{headerShown: false}}
         />
         <AuthenticatedStack.Screen
           name="PostDetailPage"
-          component={PostDetailPage}
+          component={(props) => (
+            <KeyboardWrapper>
+              <PostDetailPage {...props} />
+            </KeyboardWrapper>
+          )}
           options={{headerShown: false}}
         />
         <AuthenticatedStack.Screen
           name="ProfilePostDetailPage"
-          component={ProfilePostDetail}
+          component={(props) => (
+            <KeyboardWrapper>
+              <ProfilePostDetail {...props} />
+            </KeyboardWrapper>
+          )}
           options={{headerShown: false}}
         />
         <AuthenticatedStack.Screen
           name="OtherProfilePostDetailPage"
-          component={OtherProfilePostDetail}
+          component={(props) => (
+            <KeyboardWrapper>
+              <OtherProfilePostDetail {...props} />
+            </KeyboardWrapper>
+          )}
           options={{headerShown: false}}
         />
         <AuthenticatedStack.Screen
           name="ChannelScreen"
-          component={ChannelScreen}
+          component={(props) => (
+            <KeyboardWrapper>
+              <ChannelScreen {...props} />
+            </KeyboardWrapper>
+          )}
           options={{headerShown: false}}
         />
         <AuthenticatedStack.Screen
           name="WebsocketResearchScreen"
-          component={WebsocketResearchScreen}
+          component={(props) => (
+            <KeyboardWrapper>
+              <WebsocketResearchScreen {...props} />
+            </KeyboardWrapper>
+          )}
           options={{headerShown: false}}
         />
         <AuthenticatedStack.Screen
           name="SampleChatScreen"
-          component={SampleChatScreen}
+          component={(props) => (
+            <KeyboardWrapper>
+              <SampleChatScreen {...props} />
+            </KeyboardWrapper>
+          )}
           options={{headerShown: false}}
         />
         <AuthenticatedStack.Screen
           name="SampleChatInfoScreen"
-          component={SampleChatInfoScreen}
+          component={(props) => (
+            <KeyboardWrapper>
+              <SampleChatInfoScreen {...props} />
+            </KeyboardWrapper>
+          )}
           options={{headerShown: false}}
         />
         <AuthenticatedStack.Screen
           name="SignedChatScreen"
-          component={SignedChatScreen}
+          component={(props) => (
+            <KeyboardWrapper>
+              <SignedChatScreen {...props} />
+            </KeyboardWrapper>
+          )}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The addition of the `KeyboardWrapper` component around each screen component is a good way to ensure that the keyboard does not overlap the content on each screen. However, this approach introduces a lot of repetition in the code. Consider creating a higher-order component (HOC) that automatically wraps any given component with `KeyboardWrapper`. This would make the code more DRY (Don't Repeat Yourself) and easier to maintain.

Here's an example of how you could create such a HOC:

```javascript
function withKeyboardWrapper(Component) {
  return function WrappedComponent(props) {
    return (
      <KeyboardWrapper>
        <Component {...props} />
      </KeyboardWrapper>
    );
  };
}
```

Then, you can use this HOC when defining your screen components:

```diff
-          <AuthenticatedStack.Screen
-            name="TermsAndCondition"
-            component={(props) => (
-              <KeyboardWrapper>
-                <TermsAndCondition {...props} />
-              </KeyboardWrapper>
-            )}
-            options={{headerShown: false}}
-          />
+          <AuthenticatedStack.Screen
+            name="TermsAndCondition"
+            component={withKeyboardWrapper(TermsAndCondition)}
+            options={{headerShown: false}}
+          />
```

You would need to apply this change for each `AuthenticatedStack.Screen` in the provided code.


<!-- This is an auto-generated comment by OSS CodeRabbit -->